flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -1067,15 +1085,19 @@ static void setJobManagerAddressInConfig(Configuration config, InetSocketAddress
 	 * @param commandLine The input to the command-line.
 	 * @return custom command-line which is active (may only be one at a time)
 	 */
-	public CustomCommandLine getActiveCustomCommandLine(CommandLine commandLine) {
+	public CustomCommandLine validateAndGetActiveCommandLine(CommandLine commandLine) {
 		LOG.debug("Custom commandlines: {}", customCommandLines);
 		for (CustomCommandLine cli : customCommandLines) {
 			LOG.debug("Checking custom commandline {}, isActive: {}", cli, cli.isActive(commandLine));
 			if (cli.isActive(commandLine)) {
 				return cli;
 			}
 		}
-		throw new IllegalStateException("No command-line ran.");
+
+		throw new IllegalStateException(
+				"No valid command-line found. If you were targeting a Yarn cluster, please make sure to have " +
+				"hadoop in your classpath. For more information refer to the \"Deployment &  Operations\" section " +
```

Comment:
```suggestion
				"Hadoop in your classpath. For more information refer to the \"Deployment &  Operations\" section " +
```
maybe also say sth about`HADOOP_CLASSPATH` here.

Reply:
Actually this is wrong now because we load the `FallbackYarnSessionCLI` so if we reach here, it is not simply because of the classpath.